jlm/llvm/opt/alias-analyses/PointerObjectSet.cpp

```diff
@@ -1297,6 +1298,31 @@ PointerObjectConstraintSet::PerformOfflineVariableSubstitution()
       unificationRoot[equivalenceSetLabel] = i;
   }
 
+  // If hybrid cycle detection is enabled, it requires some information to be kept from OVS
+  if (storeRefCycleUnificationRoot)
+  {
+    // For each ref node that is in a cycle with regular node, store it for hybrid cycle detection
+    // The idea: Any pointee of p should be unified with a, if *p and a are in the same SCC
+    // NOTE: We do not use equivalence set labels here, as they represent more than just cycles
+
+    // First find one unification root representing each SCC
+    std::vector<std::optional<PointerObjectIndex>> unificationRootPerSCC(numSccs, std::nullopt);
+    for (PointerObjectIndex i = 0; i < Set_.NumPointerObjects(); i++)
+    {
+      if (!unificationRootPerSCC[sccIndex[i]])
+        unificationRootPerSCC[sccIndex[i]] = Set_.GetUnificationRoot(i);
+    }
+
+    // Assign unification roots to ref nodes that are in CYCLES with regular nodes
```

Comment:
Why is cycles written with capital letters?

Reply:
I had a bug where I assigned based on equivalence set label, so when I figured out the bug I was probably a bit eager to change the comment. Will calm it down